scripts/register_saft_addresses.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Register SAFT accounts script
+ *
+ * ts-node scripts/register_saft_addresses.ts "network" "path to csv with addresses"
+ */
+
+import fs from 'fs'
+import { utils, constants, providers, Wallet } from 'ethers'
+import { TimeLockRegistry } from '../build/types/TimeLockRegistry'
+import { TimeLockRegistryFactory } from '../build/types/TimeLockRegistryFactory'
+import { TrustToken } from '../build/types/TrustToken'
+import { TrustTokenFactory } from '../build/types/TrustTokenFactory'
+
+export const txnArgs = { gasLimit: 2_000_000, gasPrice: 20_000_000_000 }
+
+export interface SaftAccount {
+  address: string,
+  amount: string,
+}
+
+const toTrustToken = (amount: string) => utils.parseUnits(amount, 8)
+const sum = (numbers: utils.BigNumber[]) => numbers.reduce((cumSum, value) => cumSum.add(value), constants.Zero)
+
+export const registerSaftAccounts = async (registry: TimeLockRegistry, trustToken: TrustToken, accounts: SaftAccount[]) => {
+  const totalAllowance = sum(accounts.map(({ amount }) => toTrustToken(amount)))
+  const tx = await trustToken.approve(registry.address, totalAllowance, txnArgs)
+  await tx.wait()
+  console.log('Transfers approved')
+  let { nonce } = tx
+  const pendingTransactions = []
+  for (const { address, amount } of accounts) {
+    pendingTransactions.push((await registry.register(address, toTrustToken(amount), { ...txnArgs, nonce: nonce + 1 })).wait()
+      .then(() => console.log(`Done: ${address} for ${amount} TRU`))
+      .catch((err) => console.error(`Failed for ${address}`, err)),
+    )
+    nonce++
+  }
+  await Promise.all(pendingTransactions)
+}
+
+export const parseAccountList = (text: string): SaftAccount[] =>
+  text
+    .split('\n')
+    .filter((line) => line.split(',').length > 1)
+    .map((line) => ({
+      address: line.split(',')[0].trim(),
+      amount: line.split(',')[1].trim(),
+    }))
+
+const readAccountList = (filePath: string) => parseAccountList(fs.readFileSync(filePath).toString())
+
+if (require.main === module) {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const deployedAddresses = require(`./deploy/${process.argv[2]}.json`)
```

Comment:
Why not read it with `fs`?

Reply:
require also does JSON parsing for us

Reply:
It does, but it also forces us to add linter exception. Reading all files in the same manner and then parsing them imo would be prettier.